server/fleet/errors.go

```diff
@@ -289,6 +289,20 @@ func (e passwordResetRequiredError) StatusCode() int {
 	return http.StatusUnauthorized
 }
 
+// MDMNotConfiguredError is used when an MDM endpoint or resource is accessed
+// without having MDM correctly configured.
+type MDMNotConfiguredError struct{}
+
+// Status implements the transport.statuser interface so we can customize the
+// HTTP status code of the response returning this error.
+func (e MDMNotConfiguredError) Status() int {
+	return http.StatusInternalServerError
```

Comment:
Nit, but shouldn't this be a caller error instead of server? Nothing _failed_ on the server, it's just that an endpoint that hasn't been configured properly was called by a client? E.g. a general 400 bad request? Or (looking at less frequently used status codes) "501 Not Implemented" so it is easy to  differentiate in the logs between e.g. the database failed for some reason vs this case where nothing failed in a "technical way"?

Reply:
That's a good point, I thought about a `4xx` but somebody on the internet made a point that `4xx` errors need can be fixed by the client, so I switched it to a `5xx`.

In our particular case a `400` sounds good to me if it sounds good to you!

Reply:
It's just that 500 errors may very well trigger techincal support tickets/investigation of what failed, etc. (like our P1), and in this case it's not such a problem, it's just somebody / a client that hit an endpoint that isn't configured for consumption. The "client-fixable" rationale makes sense, and I think it still applies, it's "don't call that endpoint that isn't configured to work", e.g. a bit like 403 Forbidden, there's not much the client can do about it but it's not a server error. I mean, it could almost be a 404 with the more specific message of mdm not configured.

Reply:
that's a great way to put it. I changed it to be a `400`, and also the tests to use `ErrMDMNotConfigured.StatusCode()` from now on so it should be easy to change if we want to tweak it to a more specific `4xx` request